### ISSUE TITLE
Running beaverhabits with a reverse proxy

### ISSUE THREAD

**Issue by lokesh-krishna** [daya0576/beaverhabits#33]
I'm using the default ports of `8080:8080` and have nginx setup to proxy pass `http://localhost:8080`, listening on 8097. The device beaverhabits is running on is on my tailnet so the idea is to access it from my other devices this way. However, visiting `tailscale-ip-address:8097` results in a `404 Not Found`.

What might I be doing wrong here? 

My `compose.yaml`
```yml
services:
  beaver:
    container_name: beaver
    user: "1001"
    environment:
      - FIRST_DAY_OF_WEEK=0
      - HABITS_STORAGE_USER=USER_DISK
      - MAX_USER_COUNT=1
    volumes:
      - ./beaver_data:/app/.user/
    ports:
      - 8080:8080
    image: daya0576/beaverhabits:latest
```

My nginx config:
```nginx
server {
  listen 8097;

  location / {
  proxy_pass http://0.0.0.0:8080;
  proxy_http_version 1.1;
  proxy_set_header Connection "upgrade";
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header Host $host;
  }
}
```

```
Attaching to beaver                                                                                                        
beaver  | Starting Uvicorn server in production mode...                                                                    
beaver  | NICEGUI_STORAGE_PATH not set. Using default value: .user/.nicegui                                                
beaver  | INFO:     Starting BeaverHabits...                                                                               
beaver  | INFO:     Started server process [9]                                                                             
beaver  | INFO:     Waiting for application startup.                                                                       
beaver  | INFO:     Application startup complete.                                                                          
beaver  | INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)                                          
beaver  | INFO:     172.20.0.1:43150 - "GET / HTTP/1.1" 307 Temporary Redirect                                             
beaver  | INFO:     172.20.0.1:43420 - "GET / HTTP/1.1" 307 Temporary Redirect                                             
```

**Comment by ocean-haiyang** [comment]
Why're you using nginx to change the port? Why not simply change the port in the docker compose? 8097:8080

**Comment by ocean-haiyang** [comment]
services:
  beaver:
    container_name: beaver
    user: "1001"
    environment:
      - FIRST_DAY_OF_WEEK=0
      - HABITS_STORAGE_USER=USER_DISK
      - MAX_USER_COUNT=1
    volumes:
      - ./beaver_data:/app/.user/
    ports:
      - 8097:8080
    image: daya0576/beaverhabits:latest

**Comment by lokesh-krishna** [comment]
While I can access other services directly with their tailscale IP and port, this doesn't seem to work for Docker services. I've always needed to proxy them with nginx. This has worked for all other Docker services though so not sure what I'm missing here.

**Comment by ocean-haiyang** [comment]
Can you provide a compose and config of one that is working for you currently? This seems like an nginx issue not a beaverhabits issue but I've had issues too in the past with nginx.

**Comment by lokesh-krishna** [comment]
Decided to just set it up on a subdomain 😅 
I can close this issue unless debugging this is of interest to anyone else.

**Comment by ocean-haiyang** [comment]
I wonder if you create a tailscale exit node which should allow you to access your docker containers via machine IP:port instead of tailscale IP.

**Comment by lokesh-krishna** [comment]
The sidecar approach?

**Comment by ocean-haiyang** [comment]
I think I'm now having a similar issue actually.

**Comment by ocean-haiyang** [comment]
I think this was solved here #42 

Can most likely be closed @daya0576
